src/main/java/com/suse/salt/netapi/calls/modules/Locate.java

```diff
@@ -0,0 +1,182 @@
+package com.suse.salt.netapi.calls.modules;
+
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.reflect.TypeToken;
+import com.suse.salt.netapi.calls.LocalCall;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * salt.modules.locate
+ * Module for using the mlocate utilities.
+ */
+public class Locate {
+
+    private static final LocalCall<List<String>> VERSION = new LocalCall<>(
+            "locate.version",
+            Optional.empty(), Optional.empty(),
+            new TypeToken<List<String>>() {
+            });
+
+    private static final LocalCall<List<String>> UPDATEDB = new LocalCall<>(
+            "locate.updatedb",
+            Optional.empty(), Optional.empty(),
+            new TypeToken<List<String>>() {
+            });
+
+    private static final LocalCall<Stats> STATS = new LocalCall<>("locate.stats",
+            Optional.empty(), Optional.empty(),
+            new TypeToken<Stats>() {
+            });
+
+    /**
+     * Locate module result object
+     */
+    public static class Stats {
+        private long files;
+        private long directories;
+        @SerializedName("bytes in file names")
+        private long fileNamesBytes;
+        @SerializedName("bytes used to store database")
+        private long databaseBytes;
+        @SerializedName("database")
+        private String databaseLocation;
+
+        public long getFiles() {
+            return files;
+        }
+
+        public long getDirectories() {
+            return directories;
+        }
+
+        public long getFileNamesBytes() {
+            return fileNamesBytes;
+        }
+
+        public long getDatabaseBytes() {
+            return databaseBytes;
+        }
+
+        public String getDatabaseLocation() {
+            return databaseLocation;
+        }
+    }
+
+    /**
+     * All possible options for "locate" method.
+     */
+    public static class LocateOpts {
+        private Map<String, Boolean> opts = new HashMap<>();
+
+        public void setBasename(Boolean basename) {
```

Comment:
You are right that it will be boxed anyway but accepting a boxed value instead of a primitive opens up the possibility to pass `null` which is not valid in this case. So it just to remove ways to misuse the API.